tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Process/ProcessStartCommon.cs

```diff
@@ -67,9 +71,164 @@ internal static Scope CreateScope(string filename, IDictionary<string, string> e
             return scope;
         }
 
-        internal static string Truncate(string value, int maxLength)
+        private static ProcessCommandStartTags PopulateTags(string filename, IDictionary<string, string> environmentVariables, bool useShellExecute, string arguments, Collection<string> argumentList)
         {
-            return (string.IsNullOrEmpty(value) || value.Length <= maxLength) ? value : value.Substring(0, maxLength);
+            // Environment variables
+            var variablesTruncated = EnvironmentVariablesScrubber.ScrubEnvironmentVariables(environmentVariables);
+            variablesTruncated = Truncate(variablesTruncated, MaxCommandLineLength, out _);
+
+            var tags = new ProcessCommandStartTags
+            {
+                EnvironmentVariables = variablesTruncated,
+                Component = "process",
+            };
+
+            // Don't populate further with command line information if shell collection is disabled
+            if (!Tracer.Instance.Settings.CommandsCollectionEnabled)
+            {
+                return tags;
+            }
+
+            if (useShellExecute)
+            {
+                // cmd.shell
+                var commandLine = filename;
+
+                // Append the arguments to the command line
+                if (!string.IsNullOrWhiteSpace(arguments))
+                {
+                    commandLine += " " + arguments;
+                }
+                else if (argumentList != null)
+                {
+                    foreach (var argument in argumentList)
+                    {
+                        commandLine += " " + argument;
+                    }
+                }
+
+                // Truncate the command line if needed
+                commandLine = Truncate(commandLine, MaxCommandLineLength, out var truncated);
+                tags.Truncated = truncated ? "true" : null;
+
+                tags.CommandShell = commandLine;
+            }
+            else
+            {
+                // cmd.exec
+                var commandExec = new List<string> { filename };
+                if (!string.IsNullOrWhiteSpace(arguments))
+                {
+                    // Arguments are provided in a raw strings, we need to lex them
+                    var split = SplitStringIntoArguments(arguments);
```

Comment:
As an optimization, you can pass MaxCommandLineLength to this method and stop analyzing parameters when exceeded. Even you could consider truncating the arguments value before calling this method.

Reply:
Good idea 👍  I'll do it